### PR TITLE
Richer API responses for USERS and RPs

### PR DESCRIPTION
**Pull request by otoolep**

### CODE REVIEW

File: server.go [hunk 1]
```diff
@@ -41,7 +41,7 @@ const (
 	DefaultShardDuration = 7 * (24 * time.Hour)
 
 	// DefaultShardRetention is the length of time before a shard is dropped.
-	DefaultShardRetention = time.Duration(0)
+	DefaultShardRetention = 7 * (24 * time.Hour)
```
Comment (otoolep) [inline server.go]: This seemed wrong. Why have a default shard retention of 0? So I made it a week.

File: httpd/handler_test.go [hunk 2]
```diff
@@ -146,7 +146,7 @@ func TestHandler_RetentionPolicies(t *testing.T) {
 
 	if status != http.StatusOK {
 		t.Fatalf("unexpected status: %d", status)
-	} else if body != `{"results":[{"rows":[{"columns":["Name"],"values":[["bar"]]}]}]}` {
```
Comment (otoolep) [inline httpd/handler_test.go]: Do we really want capitalized column names? It wasn't clear to me. Easy to revert if so.

Comment (pauldix) [inline httpd/handler_test.go]: we don't

**Comment by pauldix** [comment]
LGTM. I'm not totally sure about having privileges returned as a column. If it's going to be more SQL ish, you should have a query to `SHOW PRIVILEGES ON <user>` and have it return results there.

**Comment by otoolep** [comment]
I was on the fence about the `privileges` column. Let me pull it, and then merge. A distinct command does seem more coherent.

**Pull request merged**